frontend/packages/git-service/src/services/bitbucket-service.ts

```diff
@@ -109,6 +109,20 @@ export class BitbucketService extends BaseService {
     }
   };
 
+  isServerURL = (isServer: boolean, specificPath?: string) => {
+    let url = '';
+    if (specificPath) {
+      url = isServer
+        ? `${this.baseURL}/projects/${this.metadata.owner}/repos/${this.metadata.repoName}/files/${this.metadata.contextDir}/${specificPath}?limit=50&at=${this.metadata.defaultBranch}`
+        : `${this.baseURL}/repositories/${this.metadata.owner}/${this.metadata.repoName}/src/${this.metadata.defaultBranch}/${this.metadata.contextDir}/${specificPath}?pagelen=50`;
+    } else {
+      url = isServer
+        ? `${this.baseURL}/projects/${this.metadata.owner}/repos/${this.metadata.repoName}/files/${this.metadata.contextDir}?limit=50&at=${this.metadata.defaultBranch}`
+        : `${this.baseURL}/repositories/${this.metadata.owner}/${this.metadata.repoName}/src/${this.metadata.defaultBranch}/${this.metadata.contextDir}?pagelen=50`;
```

Comment:
@Lucifergene Can you please explain what you are trying to achieve here?

Reply:
So, here in the case of Bitbucket, we have 2 different URL formats for fetching contents of a repository, based on the `isServer` value. 
Now for each of these URLs, there will exist again 2 cases, one having the `specificURL` in the URL, and the other not having it, based on whether `specificURL` is passed. 
Therefore, in total, there exists 4 different cases, which leads to the above code.